### PR TITLE
Fix Axon logging to use correct logger, add logging details, log changeset timedelta

### PR DESCRIPTION
**Pull request by 13rac1**

### CODE REVIEW

File: notify_listener.go [hunk 1]
```diff
@@ -32,6 +33,13 @@ func StartFromTimestamp(t time.Time) NotifyOption {
 	}
 }
 
+// NotifyLogger is an option for setting the logger
+func NotifyLogger(logger logrus.Entry) NotifyOption {
```
Comment (bernielomax) [inline notify_listener.go]: Nit: Why not just pass the pointer?

File: axon.go [hunk 2]
```diff
@@ -125,7 +125,8 @@ func (a *Axon) Run() error {
 	}
 
 	// Create a notify listener and start from the configured changeset id.
-	listener := NewNotifyListener(StartFromID(a.Config.StartFromID))
+	listenerLog := a.Logger.WithFields(logrus.Fields{"component": "listener"})
```
Comment (bernielomax) [inline axon.go]: Adding the component field could be done "by default" inside the listener.

**Review by bernielomax — COMMENTED** [review]
I think this looks ok, but I think we should hold off merging it until we confirm https://github.com/udacity/warp-pipe/pull/32 fixes sequential id issues, etc. Happy for this to be run during the user-api upgrade though.

File: axon.go (hunk 2 above)
Comment (13rac1) [reply axon.go]: Ah true, at the start.

File: notify_listener.go (hunk 1 above)
Comment (13rac1) [reply notify_listener.go]: Entirely: 😪

Comment (13rac1) [reply notify_listener.go]: fixed.

File: axon.go (hunk 2 above)
Comment (13rac1) [reply axon.go]: done

File: axon_sql.go [hunk 3]
```diff
@@ -168,7 +169,7 @@ func (a *Axon) insertRow(sourceDB *sqlx.DB, targetDB *sqlx.DB, change *Changeset
 
 	// NOTE: row insert/update/delete logs have been updated to be the same length
 	// to align timestamps to ease viewing.
-	a.Logger.Printf("row insert: %s", change)
+	a.Logger.WithField("time-delta", time.Now().Sub(change.Timestamp)).Printf("row insert: %s", change)
```
Comment (svysali) [inline axon_sql.go:172]: nit: "Lag" is more meaningful than time-delta

**Review by svysali — APPROVED** [review]
Looks good to me!

File: axon_sql.go (hunk 3 above)
Comment (13rac1) [reply axon_sql.go:172]: Same same!

**Pull request merged**